tests/Route/DefaultRouteGeneratorTest.php

```diff
@@ -164,6 +164,10 @@ public function testGenerateUrlChild(string $type, string $expected, string $nam
         $collection->add('foo');
         $collection->addCollection($childCollection);
 
+        $nochildCollection = new RouteCollection('base.Code.Child', 'admin_child', '/foo/', 'BundleName:ControllerName');
+        $nochildCollection->add('bar');
+        $collection->addCollection($nochildCollection);
```

Comment:
This variable is used nowhere so this code is useless and not testing the changes you made

Reply:
`$collection` is used in line 81. And without the changes in `DefaultRouteGenerator` the test fails after adding these 3 lines. If you want I can split the PR in 2 commits: first adding the test (which will fail) and the commit which will fix the test?

Reply:
Indeed the test failing, but in a kinda wrong way IMHO.

Can you update the switch in the test to add 
```
case 'admin_child_bar':
    return sprintf('/bar%s', $params);
```
This way you get a Failure instead of an error without your change.

Reply:
I have added this to the test. Also I rebased so the Psalm fix is in there.